clang/test/Frontend/sycl-aspects-usage.cpp

```diff
@@ -20,12 +22,18 @@ struct [[__sycl_detail__::__uses_aspects__(aspect::fp16)]] Struct {
   int a = 0;
 };
 
-[[sycl::device_has(aspect::fp64)]] int checkStructUsesAspect(int) {
+int funcWithStruct(int) {
   Struct s;
   s.a = 1;
   return s.a;
 }
 
+[[sycl::device_has(aspect::fp64)]] int checkStructUsesAspect(int a) {
+  return funcWithStruct(a);
+}
+
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -emit-llvm %s -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-EMPTY
+// CHECK-EMPTY-NOT: checkEmptyDiveceHas
```

Comment:
I'm not sure what you are testing here. Can you please explain? Are you trying to test function is not generated?

Reply:
I was testing here that this function is not present in warning messages. I will fix it right now.

Reply:
Oh I see. You were testing warning was not being generated for empty argument list. I don't think you need to test that since that comes under semantic tests of `device_has` attribute which has already been tested in FE. 

I think in this PR you just need to test propagation and usage of aspects. So for a function with `device_has` attribute with empty argument list, a good test will be to use a type with aspect in that function to ensure warning is generated